### DISCUSSION TITLE
How could I customize the tooltip on the line selection button?

### DISCUSSION THREAD

**Discussion by feasgal** [squidfunk/mkdocs-material#5195] [Q&A] [answered]
I'd like to put a different tooltip than `Toggle line selection` on the toggle button for line selection in code blocks. Before I override the `title` attribute via `extra.js`, can anyone think of a better way into that attribute?

**Comment by squidfunk** [comment, marked as answer]
We didn't localize it yet (and make it changeable), because I'm not sure if the current solution is ideal, and wanted to collect some feedback first 😊 Thanks for test-driving the new feature! How would you imagine to localize it? What do you want to change?

**Reply by feasgal** [reply]
I think the feature has a lot of potential for us, for Support being able to link directly to the lines they want to draw a user's attention to.

The tooltip for the toggle should definitely be consistent site-wide, so I was thinking something like this in `mkdocs.yml`:

```
theme:
  features:
    - content.code.select.title: My splendidly customized line select tooltip
```

At first I thought it should be a sub-item of `content.code.select` in the config, but since there's the opt-in for individual code blocks even when the toggle is not enabled site-wide, you'd still want to be able to customize the tooltip even if you didn't have `content.code.select` in your config.

**Reply by feasgal** [reply]
As far as other things I would change, only two things spring to mind, and the second one might be more appropriately handled by extra CSS as I currently am.

- I'd ***really*** like to be able to change my mind about highlighting any lines at all, without reloading the page. Currently, once I toggle the line selection on and select a line, I can only move the selection, not remove it. I'm not sure what that would look like for UI, but my instinct was to click an already-selected line.

- I used extra CSS to move the toggle and copy buttons to the left, and then padded the left side of the `code` element so the buttons don't cover the lines or the box-shadow on selected lines. I don't know whether that's going to be a common enough wish among your users to bother making right/left configurable, but it's the only other thing I'm currently modifying for our use.

**Reply by squidfunk** [reply]
Thanks, that's great feedback! It's also the exact intended use case this feature was build for. The title will be adjustable via the regular translation system – that's the most flexible approach, allows us to provide a sensible default per language, and we're building on something that has proven to work. Regarding your other remarks:

> I'd really like to be able to change my mind about highlighting any lines at all, without reloading the page. Currently, once I toggle the line selection on and select a line, I can only move the selection, not remove it. I'm not sure what that would look like for UI, but my instinct was to click an already-selected line.

Yes, fully agree. You can open an issue for this if you like.

> I used extra CSS to move the toggle and copy buttons to the left, and then padded the left side of the code element so the buttons don't cover the lines or the box-shadow on selected lines. I don't know whether that's going to be a common enough wish among your users to bother making right/left configurable, but it's the only other thing I'm currently modifying for our use. Regarding 

Positioning must be done with extra CSS, as it can be easily adjusted by the author.
